chainer/functions/loss/softmax_cross_entropy.py

```diff
@@ -220,9 +234,54 @@ def backward_gpu(self, inputs, grad_outputs):
         return gx, None
 
 
+def _double_backward_softmax_cross_entropy(x, t, normalize, class_weight,
+                                           ignore_label, reduce):
+    if isinstance(t, variable.Variable):
+        t = t.data
+
+    _check_class_weight_option(class_weight)
+    _check_reduce_option(reduce)
+    if chainer.is_debug():
+        _check_input_values(x, t, ignore_label)
+
+    loss = -chainer.functions.log_softmax(x)
+
+    if class_weight is not None:
+        shape = [1 if d != 1 else -1 for d in six.moves.range(x.ndim)]
+        class_weight = chainer.functions.broadcast_to(
+            class_weight.reshape(shape), x.shape)
+        loss = loss * class_weight
+
+    in_use = (t != ignore_label).astype(x.dtype)
+
+    loss = chainer.functions.rollaxis(loss, 1, loss.ndim)
+    loss = chainer.functions.reshape(loss, (-1, loss.shape[-1]))
+
+    # As F.select_item does not allow the elements of t to be negative,
+    # we substitute ignore_label with 0, which is valid.
+    t = t.copy()
+    t[t == ignore_label] = 0
+
+    loss = chainer.functions.select_item(loss, t.ravel())
+    loss = chainer.functions.reshape(loss, t.shape)
+
+    loss = loss * in_use
+
+    if reduce == 'mean':
+        if normalize:
+            count = in_use.sum()
+        else:
+            count = len(x)
+        count = max(count, 1.)
+        loss = loss / count
+        return chainer.functions.sum(loss)
+    else:
+        return loss
+
+
 def softmax_cross_entropy(
         x, t, normalize=True, cache_score=True, class_weight=None,
-        ignore_label=-1, reduce='mean'):
+        ignore_label=-1, reduce='mean', enable_double_backprop=True):
```

Comment:
It's better to keep the current performance by default, so let's use `False` as the default value. It would be better to add a note to the docstring of `L.Classifier` on how to turn it on (e.g. by passing a lambda function to `lossfun`).

Reply:
Done